Gvrf controls sample

Maginador opened the pull request:
GVRf joystick controls sample created by Samsung SIDIA.

GearVRf-DCO-1.0-Signed-off-by: Ricardo Maginador
ricardo.cm@samsung.com

thomasflynn commented on GVRf/Sample/gvrf_controls_sample/AndroidManifest.xml:
```diff
@@ -0,0 +1,49 @@
+        android:icon="@drawable/gearvr_logo"
+        android:label="@string/app_name"
+        android:theme="@style/AppTheme" >
+           <!--  <meta-data android:name="com.samsung.android.vr.application.mode" android:value="vr_only"/> -->
```
please uncomment this.  thanks.

thomasflynn commented on GVRf/Sample/gvrf_controls_sample/res/values/strings.xml:
```diff
@@ -0,0 +1,25 @@
+
+<resources>
+
+    <string name="app_name">GVR Simple Sample</string>
```
please call this something other than "simple sample".

thomasflynn commented:
Nice.  Here's a few things I noticed.  I haven't looked into the code to determine if any of these are intentional:

1) All the buttons on the controller light up when pressed except R1, Start, and Select.
2) The little character moves up/down/left/right with the DPad, but only moves up and left with the analog sticks.

Maginador commented:
Fixed the issues.

thomasflynn commented:
excellent.  one last question:  is the controller supposed to be behind the viewer when they start (meant as a surprise?) or is it meant to be in front?

Maginador commented:
That's the expected behavior. So the user will find new objects as he looks around.

thomasflynn commented:
ok, cool.  looked at the code.  it seems this commit also includes the keyboard sample.  was that intended??

Maginador commented:
No, both samples were in the same branch, Keyboard was sent accidentally with Control.

thomasflynn commented:
so, you're going to remove it, right?

Maginador commented:
Yes I'll remove it as soon as possible and give the feedback here.

chenchao1407 commented on GVRf/Sample/gvrf_controls_sample/assets/gvr_note4.xml:
```diff
@@ -0,0 +1,21 @@
+
+<lens name="N4">
+    <scene fov-y="90.0" camera-separation-distance="0.062" fbo-width="1024" fbo-height="1024" msaa="1"/>
+</lens>
```
We just changed the format of xml configurations. Please follow other samples to see how it is set now.

Thanks in advance.

chenchao1407 commented on GVRf/Sample/gvrf_controls_sample/src/org/gearvrf/controls/ChainElement.java:
```diff
@@ -0,0 +1,51 @@
+
+        getTransform().setPosition(
+                currentX + nextX, currentY + nextY, currentZ + nextZ);
+
```
Not quite sure why you need to do the computation.

currentX + nextX = target.getTransform().getPositionX() - currentX + currentX = target.getTransform().getPositionX().

Same with currentY + nextY, currentZ + nextZ.

chenchao1407 commented on GVRf/Sample/gvrf_controls_sample/src/org/gearvrf/controls/util/VRSamplesTouchPadGesturesDetector.java:
```diff
@@ -0,0 +1,278 @@
+//            Log.d(DEBUG_TAG, " - " + sharedPref.getFloat("SWIPE_DISTANCE"+i, -00000));
+//        }
+//    }
+
```
remove commented codes, thanks

chenchao1407 commented:
Tried with controls sample without a controller, looks fine to me. And still waiting for you to remove the keyboard sample.

Thanks

Maginador commented:
Removed Keyboard commit from this branch, changed xml structure as asked, removed commented code and removed ChainElement.java, it is not used inside the project.
